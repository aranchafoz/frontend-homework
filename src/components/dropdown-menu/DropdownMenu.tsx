import { connect } from 'react-redux'
import { createRef, RefObject } from 'react'


import { toggleDropdownMenuVisibility } from '../../redux/actions'

import MenuLink from '../menu-link/MenuLink'
import Companies from '../companies/Companies'
import { useOutsideClick } from '../../hooks/use-click-outside';

type DispatchProps = {
  toggleDropdownMenuVisibility: () => void,
}

const DropdownMenu = ({ toggleDropdownMenuVisibility }: DispatchProps) => {

  const dropdownRef: RefObject<any> = createRef();
  useOutsideClick(dropdownRef, () => toggleDropdownMenuVisibility());

  function handleLinkClick() {
    toggleDropdownMenuVisibility()
  }

  return (
    <div ref={dropdownRef} className="nav__menu">
      <Companies />

      <div>
        <MenuLink
          text="Get the mobile app"
          icon="phone_iphone"
          onClick={handleLinkClick}
        />

        <MenuLink
          text="Community"
          icon="people"
          onClick={handleLinkClick}
        />

        <MenuLink
          text="Knowledge base"
          icon="book"
          onClick={handleLinkClick}
        />
      </div>

      <div>
        <MenuLink
          text="Settings"
          icon="settings"
          onClick={handleLinkClick}
        />

        <MenuLink
          type="logout"
          text="Log out"
          icon="exit_to_app"
          onClick={handleLinkClick}
        />
      </div>
    </div>
  )
}

export default connect(
  null,
  { toggleDropdownMenuVisibility }
)(DropdownMenu)