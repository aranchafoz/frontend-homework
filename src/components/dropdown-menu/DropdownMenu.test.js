import { shallow } from 'enzyme'
import DropdownMenu from './DropdownMenu'

describe('<DropdownMenu />', () => {
  const toggleDropdownMenuVisibilityMock = jest.fn()

  const render = () => shallow(
    <DropdownMenu
      toggleDropdownMenuVisibility={toggleDropdownMenuVisibilityMock}
    />
  )
  it('renders menu', () => {
    expect(render()).toMatchSnapshot()
  })
})
