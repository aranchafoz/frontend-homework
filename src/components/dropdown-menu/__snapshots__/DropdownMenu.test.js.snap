// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownMenu /> renders menu 1`] = `
<div>
  <Connect(Companies) />
  <div>
    <MenuLink
      icon="phone_iphone"
      text="Get the mobile app"
    />
    <MenuLink
      icon="people"
      text="Community"
    />
    <MenuLink
      icon="book"
      text="Knowledge base"
    />
  </div>
  <div>
    <MenuLink
      icon="settings"
      text="Settings"
    />
    <MenuLink
      icon="exit_to_app"
      text="Log out"
    />
  </div>
</div>
`;
